### PR TITLE
Rlabkey v3.0.0

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
Issue [41677](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=41677): Fix for domain get/save round tripping of domain indices (need to use list() for index columnNames)

Issue [48561](https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=48561): Update R API to encode SQL parameters

#### Related Pull Requests
* https://github.com/LabKey/platform/pull/4699
* https://github.com/LabKey/platform/pull/4879

#### Changes
* add labkey.setWafEncoding() helper and use wafEncoding() in labkey.executeSql()
* labkey.domain.get to cast indices columnNames to list() instead of character array

### CODE REVIEW

File: Rlabkey/NEWS [hunk 1]
```diff
@@ -1,3 +1,10 @@
+Changes in 2.13.0
+  o Issue 41677: Fix for domain get/save round tripping of domain indices (need to use list() for index columnNames)
+  o Obfuscate SQL parameters passed by labkey.executeSql to avoid rejection by web application firewalls
```
Comment (labkey-jeckels) [inline Rlabkey/NEWS]:
```suggestion
  o Encode SQL parameters passed by labkey.executeSql to avoid rejection by web application firewalls
```

**Review by labkey-jeckels — APPROVED** [review]
Code changes look good, but a number of tests failed. Please review and address as appropriate.

**Review by labkey-klum — APPROVED** [review]

**Comment by labkey-adam** [comment]
Susan & I talked about publishing the newest Java remote API version as a new major release (v6.0.0) since it's not backward compatible with existing servers. Do we want to do the same thing here?

**Comment by cnathe** [comment]
> Susan & I talked about publishing the newest Java remote API version as a new major release (v6.0.0) since it's not backward compatible with existing servers. Do we want to do the same thing here?

Good idea. I'll plan to do that.

First I have to figure out the `RlabkeyTest.testRlabkeyQueryApi` failure. Apparently, the set of encoded characters is different between the JS encodeURIComponent function and the R URLencode function.

**Comment by labkey-adam** [comment]
> > Susan & I talked about publishing the newest Java remote API version as a new major release (v6.0.0) since it's not backward compatible with existing servers. Do we want to do the same thing here?
> 
> Good idea. I'll plan to do that.
> 
> First I have to figure out the `RlabkeyTest.testRlabkeyQueryApi` failure. Apparently, the set of encoded characters is different between the JS encodeURIComponent function and the R URLencode function.

In the Java API, I implemented the encoding to match what we're doing on the server side: https://github.com/LabKey/labkey-api-java/blob/develop/src/org/labkey/remoteapi/internal/EncodeUtils.java

Maybe it's the "+" to "%20" replacement?

**Pull request merged**